modules/mono/csharp_script.cpp

```diff
@@ -2664,6 +2664,9 @@ Error CSharpScript::load_source_code(const String &p_path) {
 
 #ifdef TOOLS_ENABLED
 	source_changed_cache = true;
+	if (ResourceLoader::get_timestamp_on_load()) {
+		set_last_modified_time(FileAccess::get_modified_time(p_path));
+	}
 #endif
```

Comment:
I'm not sure which one is correct, but I noticed a difference with GDScript.

https://github.com/godotengine/godot/blob/b8ed596769d2114797015833ce86f86ee872ecfa/modules/gdscript/gdscript.cpp#L1049-L1053

Reply:
`ResourceLoader::get_timestamp_on_load()` will always return `true` in `TOOLS_ENABLED`, so it is not necessary.

Reply:
Could you please also see if `set_edited(false);` is needed? I do not insist, I just guess that this may be related.

Reply:
For `CSharpScript`, modifying the source code before compilation has no effect.

I'm not sure what `set_edited()` does. I'm curious why it only exists in `GDScript::load_source_code()` but not in `GDScript::set_source_code()`.

`ResourceLoader::load()` usually returns a new instance. And this PR aims to get the text from the file without changing the instance. Used to simplify the following case:

https://github.com/godotengine/godot/blob/23318e877890029f35856036b9c4e0bfa09cacc6/editor/plugins/script_editor_plugin.cpp#L2579-L2582

Well, if the gd file is deleted externally, the popup will only appear once, and the cache may not have been deleted.
 GDScript needs to update its cache.